packages/core-flows/src/definition/cart/steps/find-sales-channel.ts

```diff
@@ -0,0 +1,34 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { ISalesChannelModuleService, SalesChannelDTO } from "@medusajs/types"
+import { MedusaError } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  salesChannelId?: string
+}
+
+export const findSalesChannelStepId = "find-sales-channel"
+export const findSalesChannelStep = createStep(
+  findSalesChannelStepId,
+  async (data: StepInput, { container }) => {
+    const salesChannelService = container.resolve<ISalesChannelModuleService>(
+      ModuleRegistrationName.SALES_CHANNEL
+    )
+
+    let salesChannel: SalesChannelDTO | undefined
+    if (data.salesChannelId) {
+      salesChannel = await salesChannelService.retrieve(data.salesChannelId)
+    } else {
+      // TODO: Find default sales channel from store
+    }
```

Comment:
thought(non-blocking): we could even just make it a requirement that the sales channel is passed to the workflow and avoid any fallback. But let's revisit the question when we have publishable keys.